### ISSUE TITLE
Realtime safe containers (for VST3 wrapper)

### ISSUE THREAD

**Issue by 0ax1** [free-audio/clap-wrapper#134]
When using insert/delete operations, containers from the C++ standard library should never be considered real-time safe, even if `reserve` is called on construction. This is in particular relevant for ~`std::vector`~ and `std::unordered_map`, as they tend to be commonly used. 

*Correction: Opposed to `unordered_map`, `vector` insert and delete operation within capacity limits seem to be realtime safe.*

The effects can be observed e.g. on macOS when running dtrace: 
```
// file: dtrace_rt.d 

// Trace the following function calls on real-time threads, with thread priorities higher than 90 (0x5a).
pid$target::*_alloc*:entry / curthread->sched_pri > 0x5a / { @[ustack()] = count(); }
pid$target::*malloc*:entry / curthread->sched_pri > 0x5a / { @[ustack()] = count(); }
pid$target::*pthread*:entry / curthread->sched_pri > 0x5a / { @[ustack()] = count(); }
pid$target::*_lock*:entry / curthread->sched_pri > 0x5a / { @[ustack()] = count(); }
pid$target::objc_msgSend:entry / curthread->sched_pri > 0x5a / { @[ustack()] = count(); }
```

Tracing: ```sudo dtrace -s dtrace_rt.d -p `pgrep -x BitwigPluginHost-ARM64-NEON` > $TMPDIR/dtrace_report.log```.
Observing the callstacks: `less $TMPDIR/dtrace_report.log`.

Note that in order for dtrace to function correctly the trace should not be piped into `less` directly.

Callstacks leading to non-realtime safe API calls in context of the VST3 wrapper can be observed here:
[dtrace-vst3-wrapper.log](https://github.com/free-audio/clap-wrapper/files/12595008/rt-trace-vst3-wrapper.log)

--- 

The solution is to replace the std containers with custom ones which are implemented on top of an array. Alongside, we'll need to introduce a unit test target to make visible that the introduced containers function correctly.

**Comment by 0ax1** [comment]
will be resolved by: https://github.com/free-audio/clap-wrapper/pull/136

**Comment by defiantnerd** [comment]
Merged to 0.6.0
